src/Api/PubnubApi/RetryConfiguration.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+namespace PubnubApi
+{
+	public class RetryConfiguration
+	{
+		public IRetryPolicy RetryPolicy { get; set; }
+
+		public RetryConfiguration(IRetryPolicy retryPolicy)
+		{
+			this.RetryPolicy = retryPolicy;
+		}
+		
+		public static RetryConfiguration Linear(int delay, int maxRetry) => new RetryConfiguration(new LinearRetryPolicy(delay, maxRetry));
+
+		public static RetryConfiguration Exponential(int minDelay, int maxDelay, int maxRetry) => new RetryConfiguration(new ExponentialRetryPolicy(minDelay, maxDelay, maxRetry));
+
+	}
+
+	public interface IRetryPolicy
+	{
+		bool ShouldRetry(int attemptedRetries, PNStatus status);
+		int GetDelay(int attemptedRetries, PNStatus status, int? retryAfter);
+	}
+
+	internal class LinearRetryPolicy : IRetryPolicy
+	{
+		private int delay;
```

Comment:
Question:
Is in C# any way of handling time? Like Rust's `Duration`? If not or `int` is a common way - I'm fine with that.

Reply:
I considered that option. C# has `TimeSpan` Struct
The reason why I haven't use that is,
From User perspective, We have many other configuration where we need to specify duration/intervals. And there there is always integer value (in seconds).  So this one can be odd one. All of them are in PNConfiguration obviously.

We can discuss and change this... It can be a quick change. So It's open for change!!!

Reply:
Okay dokey. This is good explanation for me to keep it as it is.